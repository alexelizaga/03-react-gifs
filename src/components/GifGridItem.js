import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url}) => {
  return (
    <div data-testid='GifGridItem' className='card animate__animated animate__fadeIn'>
      <img data-testid='GifGridItemImg' src={url} alt={title} />
      <p data-testid='GifGridItemTitle'>{ title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}