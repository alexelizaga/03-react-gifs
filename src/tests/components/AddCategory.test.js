import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import { AddCategory } from '../../components/AddCategory';

describe('Test <AddCategory />', () => {

  test('should render <AddCategory /> correctly', () => {
    const tree = renderer
      .create(<AddCategory setCategories={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});