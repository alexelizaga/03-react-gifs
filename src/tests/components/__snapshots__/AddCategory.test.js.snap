// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddCategory /> should render <AddCategory /> correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
